mautic/mautic
# Ignore queue mode for send email action

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3828
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:
Added option to  send email immediately  even if queue email mode is enabled.

![image](https://user-images.githubusercontent.com/462477/33602648-41e7d95c-d9b0-11e7-97ef-89144cf0bbae.png)


#### Steps to test this PR:
1.  Apply PR and got to Configurations > Emails > and  Enable queue
2.  Create form with Send email to  Contact action with new Send immediately option set to No
3:  Send form, email should go to queue (verify it in directory app\spool\default\) 
4. Then Go to form and edit action to Send immediately YES
5. Send from, email should send immediately without queue  (again verify it in directory app\spool\default\ - new file shoudn't create) 
6. Then change Configurations > Emails > and change  Queue to immediately 
7. Send form and email should send immediately    (again verify it in directory app\spool\default\ - new file shoudn't create)
